server/datastore/mysql/migrations/tables/20240521143023_CreateTableMDMAssets.go

```diff
@@ -0,0 +1,38 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240521143023, Down_20240521143023)
+}
+
+func Up_20240521143023(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE mdm_config_assets (
+    id int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY, 
+
+    -- name is used for humans to identify what value is stored in this row
+    name varchar(256) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
+
+    -- value holds the raw value of the asset
+    value blob NOT NULL,
+
+    -- this table does soft deletes, and the application logic is in charge of
+    -- preventing INSERTs of two rows with the same name that are not deleted
+    deleted_at timestamp NULL DEFAULT NULL,
```

Comment:
want to call this out, the alternative I had in mind is to have a non-nullable `deleted_at` column that we could use as part of an `UNIQUE KEY (name, deleted_at)`.

I was considering:

1. Make `deleted_at` a boolean, but this will only allow one deleted row
2. Make `deleted_at` `NOT NULL` and the default value be a hardcoded date, like we do for labels and policies

Reply:
Yeah, so currently there's no unique key preventing duplicate names at all (and I'm guessing we'll want to select with a unique, single-non-deleted row `name` when loading assets for e.g. APNs, SCEP, etc.). 

The nullable `deleted_at` in a unique index does not do what we want, since it would allow multiple pairs of `(name, NULL)` (unlike, IIRC, postgresql which treats tuples with NULL as the same for unique keys).

How about adding both a `deletion_uuid` string (non-nullable, defaults to empty) AND a `deleted_at` timestamp (nullable) and have a UNIQUE index on `(name, deletion_uuid)` so that only one row is not deleted and it's easy to query for live rows (no need for date comparison)? When we delete, we add a uuid in that column and a timestamp in `deleted_at` for audit/debugging purposes. Does that make sense? Can we think of a simpler approach that would allow the unique key?

Reply:
Another concern (as is always the case with soft deletions) is to forget that the table supports them, so we query with just the `name = ?` condition and may get a deleted row. I don't have a good solution to this... (not that this is a blocker in any way, that's just how it is!)

Reply:
I like that approach @mna !

Reply:
sounds good and I agree with that part of soft deletes :( went with this approach and added a couple of tests to get a feel of it.